src/main.rs

```diff
@@ -119,40 +126,49 @@ created: {created_at}
     }
 }
 
+impl fmt::Display for CrateDependency {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{} {}", self.id, self.req)
+    }
+}
+
 /// fetches and prints package metadata from crates.io
-fn print_crate_metadata(crate_name: &str, as_json: bool) {
-    let mut reg = crates_io::Registry::new(DEFAULT.to_string(), None);
-
-    fn get_crate_metadata(crate_name: &str, body: &str) -> Option<CrateMetadata> {
-        let resp_json: Result<CrateMetaResponse, _> = serde_json::from_str(body);
-        match resp_json {
-            Ok(data) => {
-                Some(data.crate_data)
-            }
-            Err(e) => {
-                println_stderr!("Error parsing JSON data for {}: {}", crate_name, e);
-                None
-            }
-        }
+fn print_crate_metadata(crate_name: &str, as_json: bool, with_deps: bool) -> Result<(), String> {
+    let mut req = crates_io::Registry::new(DEFAULT.to_string(), None);
+
+    let response = req.get_crate_data(crate_name)
+        .map_err(|e| format!("Error fetching data for {}: {}", crate_name, e))?;
+
+    let meta: Result<CrateMetaResponse, _> = serde_json::from_str(&response)
+        .map_err(|e| format!("Error parsing JSON data for {}: {}", crate_name, e));
+
+    let meta = meta?.crate_data;
+
+    if as_json {
+        println!("{}", meta);
+        return Ok(());
     }
 
-    match reg.get_crate_data(crate_name) {
-        Ok(data) => {
-            if as_json {
-                println!("{}", data);
-                return ();
-            }
-
-            match get_crate_metadata(crate_name, &*data) {
-                Some(crate_meta) => print!("{}", crate_meta),
-                None => {}
-            }
-        }
-        Err(e) => {
-            // e.g. crate name not found
-            println_stderr!("Error fetching data for {}: {}", crate_name, e);
+    // print crate's metadata
+    println!("{}", meta);
```

Comment:
If you'd rather have quieter output, I'd be OK with just printing just the id or name or both with the dependents instead of the full metadata.